assets/js/components/user-input/UserInputQuestionnaire.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * User Input Questionnaire.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ProgressBar from '../ProgressBar';
+import UserInputQuestionWrapper from './UserInputQuestionWrapper';
+import UserInputSelectOptions from './UserInputSelectOptions';
+import UserInputKeywords from './UserInputKeywords';
+import UserInputPreview from './UserInputPreview';
+import {
+	USER_INPUT_ANSWERS_GOALS,
+	USER_INPUT_ANSWERS_HELP_NEEDED,
+	USER_INPUT_ANSWERS_POST_FREQUENCY,
+	USER_INPUT_ANSWERS_ROLE,
+} from './util/constants';
+
+export default function UserInputQuestionnaire() {
+	const [ activeSlug, setActiveSlug ] = useState( 'role' );
+
+	const questions = [ 'role', 'postFrequency', 'goals', 'helpNeeded', 'searchTerms', 'preview' ];
+	const activeSlugIndex = questions.indexOf( activeSlug );
+
+	const next = useCallback( () => {
+		setActiveSlug( questions[ activeSlugIndex + 1 ] );
+	}, [ activeSlugIndex ] );
+
+	const goTo = useCallback( ( steps = 1 ) => {
+		setActiveSlug( questions[ activeSlugIndex - steps ] );
+		global.scrollTo( 0, 0 );
+	}, [ activeSlugIndex ] );
+
+	const back = goTo.bind( null, 1 );
+
+	return (
+		<Fragment>
+			<ProgressBar
+				height={ 0 }
+				indeterminate={ false }
+				progress={ ( activeSlugIndex + 1 ) / questions.length }
```

Comment:
Currently, this is not fully correct as it also takes the `preview` step into account, but that shouldn't affect the progress bar progress. Based on my previous comment, if this relies on really only the `questions` (not `steps`, see previous comment), then it would work as expected.

Reply:
Updated.